app/goal_routes.py

```diff
@@ -0,0 +1,113 @@
+from flask import Blueprint, jsonify, abort, request, make_response
+from app import db
+from app.models.goal import Goal
+from app.models.task import Task
+
+
+
+
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+# VALIDATE FUNCTION
+
+def validate_id(cls, id):
+    try:
+        id = int(id)
+    except:
+        abort(make_response({"message":f"{cls.__name__} {id} invalid"}, 400))
+
+    obj = cls.query.get(id)
+
+    if not obj:
+        abort(make_response({"message":f"{cls.__name__} {id} not found"}, 404))
+    
+    return obj
```

Comment:
This exactly matches the definition of `validate_id` in `routes.py`, so the repetition is not necessary. As mentioned in the comment about the naming of `routes.py`, I would put this shared function in a utilities file, something like`utils.py`.